rewrite-xml/src/main/java/org/openrewrite/xml/SpringJavaTypeReferenceProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.xml;
```

Comment:
I was going to suggest to move this to an internal package, but we might even have to move it to the `tree` package for the class loader issues we've discussed. I think we will have to test this and also make sure that the class loaders can properly load the `META-INF/services` files.